bowling/src/DataParser.cpp

```diff
@@ -0,0 +1,77 @@
+#include "DataParser.hpp"
+
+#include <algorithm>
+
+std::string DataParser::stringPreparing(std::string& playerResults) {
+    const char addToEndOfString = '|';
+    playerResults.push_back(addToEndOfString);
+
+    return playerResults;
+}
+
+std::vector<std::string> DataParser::stringSplitting(std::string& preparedStringWithPlayerResults) {
+    const std::string delimiter = "|";
+    size_t position = 0;
+    std::string token;
+    std::vector<std::string> vectorWithResultsForEachLine{};
+    while ((position = preparedStringWithPlayerResults.find(delimiter)) != std::string::npos) {
+        token = preparedStringWithPlayerResults.substr(0, position);
+        vectorWithResultsForEachLine.emplace_back(token);
+        preparedStringWithPlayerResults.erase(0, position + delimiter.length());
+    }
+
+    return vectorWithResultsForEachLine;
+}
+
+std::vector<int> DataParser::stringProcessing(std::vector<std::string>& vectorWithResultsForEachLine) {
+    std::vector<int> vectorWithResults{};
+
+    std::find_if(vectorWithResultsForEachLine.cbegin(), vectorWithResultsForEachLine.cend(),
+                 [&vectorWithResults](std::string bowlingLineString) {
+                     std::find_if(bowlingLineString.cbegin(), bowlingLineString.cend(),
```

Comment:
using emplace_back in your lambdas tells me that std::find_if is not a proper algorithm for that.

Reply:
I dont quite understand why. Is the point that std::find_if is only to be used for searching? What other algorithm do you suggest? Std::find_if was the first that came to my mind :(.

Reply:
maybe some `transform / generate / copy / move / fill`? If you want to put some data into a container it is first what comes into my mind. There is also `search` algorithm.
You have nested find_if, maybe one of them is good. Maybe even both are good, but there is something "dark" that I see find_if + emplace_back :D